EcoSoapBank/Pickups/PickupsView.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  PickupsView.swift
+//  EcoSoapBank
+//
+//  Created by Jon Bash on 2020-08-10.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import SwiftUI
+
+
+protocol PickupsViewDelegate: AnyObject {
+    func logOut()
+}
+
+
+struct PickupsView: View {
+    @ObservedObject private var pickupController: PickupController
+
+    @State private var makingNewPickup = false
+
+    private weak var delegate: PickupsViewDelegate?
+
+    init(
+        pickupController: PickupController,
+        delegate: PickupsViewDelegate?
+    ) {
+        self.pickupController = pickupController
+        self.delegate = delegate
+    }
+
+    var body: some View {
+        NavigationView {
+            PickupHistoryView(pickupController: pickupController)
+                .navigationBarTitle("Pickup History", displayMode: .inline)
+                .navigationBarItems(
+                    leading: Button(
+                        action: { self.delegate?.logOut() },
+                        label: { Text("Log out") }),
+                    trailing: Button(
+                        action: { self.makingNewPickup = true },
+                        label: newPickupButtonLabel
+                    )
+                )
+        }
+    }
+
+    private func newPickupButtonLabel() -> some View {
+        HStack(spacing: 4) {
+            Image.plus()
+            Image.cubeBox()
+        }.padding(4).overlay(RoundedRectangle(cornerRadius: 10).stroke())
```

Comment:
This seems like it could maybe use a touch more padding (the top of the box touches the outer rectangle). This can be done at a later point of course, is not mission-critical right now.

Reply:
Yep, this was a button I spent far, far too much time on and decided to move on from lest I spend my entire afternoon on it...